src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java

```diff
@@ -218,6 +206,96 @@
         }
     }
 
+    private ResponseEntity<?> getListedSubscriptions(String subscriptionName) {
+        // set is used to prevent subscription names repeating
+        Set<String> subscriptionNameList = new HashSet<>(Arrays.asList(subscriptionName.split(",")));
+        List<Subscription> foundSubscriptionList = new ArrayList<>();
+        List<String> notFoundSubscriptionList = new ArrayList<>();
+
+        subscriptionNameList.forEach(name -> {
+            try {
+                LOGGER.debug("Subscription fetching has been started: {}", name);
+                // Make sure the password is not sent outside this service.
+                Subscription subscription = subscriptionService.getSubscription(name);
+                subscription.setPassword("");
+                foundSubscriptionList.add(subscription);
+                LOGGER.debug("Subscription [{}] fetched successfully.", name);
+            } catch (SubscriptionNotFoundException e) {
+                LOGGER.debug("Subscription not found: {}", name, e);
+                notFoundSubscriptionList.add(name);
+            } catch (Exception e) {
+                LOGGER.error("Failed to fetch subscription {}", name, e);
+                notFoundSubscriptionList.add(name);
+            }
+        });
+        GetSubscriptionResponse response = new GetSubscriptionResponse();
+        response.setFoundSubscriptions(foundSubscriptionList);
+        response.setNotFoundSubscriptions(notFoundSubscriptionList);
+        HttpStatus httpStatus;
+        if (!foundSubscriptionList.isEmpty()) {
+            httpStatus = HttpStatus.OK;
+        } else {
+            httpStatus = HttpStatus.NOT_FOUND;
+        }
+        if (httpStatus == HttpStatus.NOT_FOUND) {
+            String errorMessage = "Failed to fetch subscriptions:\n" + notFoundSubscriptionList.toString();
+            String errorJsonAsString = ResponseMessage.createJsonMessage(errorMessage);
+            return new ResponseEntity<>(errorJsonAsString, httpStatus);
+        }
+        return new ResponseEntity<>(response, httpStatus);
+    }
+
+    private ResponseEntity<?> deleteSingleSubscription(String subscriptionName) {
+        LOGGER.debug("Subscription deletion has started: {}", subscriptionName);
+        try {
+            if (subscriptionService.deleteSubscription(subscriptionName)) {
+                LOGGER.debug("Subscription was deleted successfully: {}", subscriptionName);
+                return new ResponseEntity<>(HttpStatus.OK);
```

Comment:
Can we have a success message when a subscription is deleted here?

Reply:
We use the http status to see if it's gone well. Same for POST and PUT.